org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporterTest.java

```diff
@@ -95,4 +97,26 @@ public void testGradlePersistence() throws Exception {
 		assertTrue(GradleProjectImporter.shouldSynchronize(project.getLocation().toFile()));
 	}
 
+	@Test
+	public void testWorkspaceSettings() throws Exception {
+		File file = null;
+		try {
+			IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
+			File rootFile = root.getLocation().toFile();
+			file = new File(rootFile, "fakeGradleHome");
+			System.setProperty(GradleProjectImporter.GRADLE_HOME, file.getAbsolutePath());
+			boolean overrideWorkspaceSettings = GradleProjectImporter.getGradleHomeFile() != null;
+			assertFalse(overrideWorkspaceSettings);
```

Comment:
fails on my machine here because I have GRADLE_HOME set as an env var. Test should make sure sysenv and sysprops are set to null before, then restored after the test.